main.go

```diff
@@ -1,235 +1,75 @@
 package main
 
 import (
-	"encoding/json"
 	goflag "flag"
 	"fmt"
-	"io/ioutil"
 	"math/rand"
 	"os"
-	"strconv"
 	"time"
 
-	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	cloudprovider "k8s.io/cloud-provider"
+	"k8s.io/cloud-provider/app"
+	cloudcontrollerconfig "k8s.io/cloud-provider/app/config"
+	"k8s.io/cloud-provider/options"
 	cliflag "k8s.io/component-base/cli/flag"
 	"k8s.io/component-base/logs"
 	_ "k8s.io/component-base/metrics/prometheus/clientgo" // for client metric registration
 	_ "k8s.io/component-base/metrics/prometheus/version"  // for version metric registration
 	"k8s.io/klog/v2"
-	"k8s.io/kubernetes/cmd/cloud-controller-manager/app"
 
 	"github.com/equinix/cloud-provider-equinix-metal/metal"
 	"github.com/spf13/pflag"
 )
 
-const (
-	apiKeyName                   = "METAL_API_KEY"
-	projectIDName                = "METAL_PROJECT_ID"
-	facilityName                 = "METAL_FACILITY_NAME"
-	loadBalancerSettingName      = "METAL_LB"
-	envVarLocalASN               = "METAL_LOCAL_ASN"
-	envVarBGPPass                = "METAL_BGP_PASS"
-	envVarAnnotationLocalASN     = "METAL_ANNOTATION_LOCAL_ASN"
-	envVarAnnotationPeerASNs     = "METAL_ANNOTATION_PEER_ASNS"
-	envVarAnnotationPeerIPs      = "METAL_ANNOTATION_PEER_IPS"
-	envVarAnnotationSrcIP        = "METAL_ANNOTATION_SRC_IP"
-	envVarAnnotationBGPPass      = "METAL_ANNOTATION_BGP_PASS"
-	envVarEIPTag                 = "METAL_EIP_TAG"
-	envVarAPIServerPort          = "METAL_API_SERVER_PORT"
-	envVarBGPNodeSelector        = "METAL_BGP_NODE_SELECTOR"
-	defaultLoadBalancerConfigMap = "metallb-system:config"
-)
-
-var (
-	providerConfig string
-)
-
 func main() {
 	rand.Seed(time.Now().UTC().UnixNano())
 
-	command := app.NewCloudControllerManagerCommand()
+	opts, err := options.NewCloudControllerManagerOptions()
+	opts.KubeCloudShared.CloudProvider.Name = metal.ProviderName
```

Comment:
This probably should be checked for various `nil`s, as `KubeCloudShared` and `CloudProvider` are pointers.

Reply:
I copied these two lines from DO. Others may have it too.  I exported ProviderName to match the pattern.

Reply:
Heh, so did I (and many others) when getting started here. 

Either way, we should be careful and check for nils.

Reply:
I dug into their (k8s) code, looks like this just is the way it is. But let's check those nils.